spark-cassandra-connector-embedded/src/main/scala/com/datastax/spark/connector/embedded/EmbeddedCassandra.scala

```diff
@@ -16,48 +16,151 @@ trait EmbeddedCassandra {
+
+object UserDefinedProperty {
+  type propertyType = NodeProperty {val propertyName: String}
+  final val hostProperty = HostProperty("IT_TEST_CASSANDRA_HOSTS")
+  final val nativePortProperty = NativePortProperty("IT_TEST_CASSANDRA_NATIVE_PORTS")
+  final val rpcPortProperty = RpcPortProperty("IT_TEST_CASSANDRA_RPC_PORTS")
+
+  val hosts = getNodePropertySeq(hostProperty).asInstanceOf[IndexedSeq[InetAddress]]
+  val nativePorts = getNodePropertySeq(nativePortProperty).asInstanceOf[IndexedSeq[Int]]
+  val rpcPorts = getNodePropertySeq(rpcPortProperty).asInstanceOf[IndexedSeq[Int]]
+
+  private def getNodePropertySeq(nodeProperty: NodeProperty): IndexedSeq[Any] = {
+    nodeProperty match {
+      case NativePortProperty(p) => getValueSeq(p).map(e => e.toInt)
+      case RpcPortProperty(p) => getValueSeq(p).map(e => e.toInt)
+      case HostProperty(p) => getValueSeq(p).map(e => InetAddress.getByName(e))
+      case _ => throw new RuntimeException("Wrong node input property")
+    }
+  }
+
+  private def getValueSeq(propertyName: String) = {
+    sys.env.get(propertyName) match {
+      case Some(p) => p.split(",").map(e => e.trim).toIndexedSeq
+      case None => IndexedSeq()
+    }
+  }
+
+  def getProperty(nodeProperty: NodeProperty) = {
+    nodeProperty match {
+      case p: propertyType => sys.env.get(p.propertyName)
+      case _ => throw new RuntimeException("Wrong node input property")
+    }
+  }
+}
+
 object EmbeddedCassandra {
 
-  val DefaultHost = "127.0.0.1"
+  import UserDefinedProperty._
+
+  getProperty(hostProperty) match {
+    case Some(h) =>
+      val hostSize = h.split(",").size
+      getProperty(nativePortProperty) match {
+        case Some(np) =>
+          val nativePortSize = np.split(",").size
+          require(hostSize == nativePortSize, "IT_TEST_CASSANDRA_HOSTS must have same size as IT_TEST_CASSANDRA_NATIVE_PORTS")
+          getProperty(rpcPortProperty) match {
+            case Some(rp) =>
+              val rpcPortSize = rp.split(",").size
+              require(hostSize == rpcPortSize, "IT_TEST_CASSANDRA_HOSTS must have same size as IT_TEST_CASSANDRA_RPC_PORTS")
+            case None => throw new RuntimeException("Missing IT_TEST_CASSANDRA_RPC_PORTS settings in system environment")
+          }
+        case None => throw new RuntimeException("Missing IT_TEST_CASSANDRA_NATIVE_PORTS settings in system environment")
+      }
+    case None =>
+  }
 
-  // TODO change to CASSANDRA_HOST
-  val HostProperty = "IT_TEST_CASSANDRA_HOST"
+  private[connector] var cassandraRunners: IndexedSeq[Option[CassandraRunner]] = IndexedSeq(None)
+
+  private[connector] var currentConfigTemplates: IndexedSeq[String] = IndexedSeq()
+
+  def getProps(index: Integer): Map[String, String] = {
+    require(hosts.isEmpty || index < hosts.size, s"$index index is overflow the size of ${hosts.size}")
+    val host = getHost(index).getHostAddress
+    Map("seeds"               -> host,
```

Comment:
I think this should be configurable somewhere - maybe not for this ticket but in the future i would be nice not to only start nodes in separate clusters but also in a single dc.